docs/DeveloperGuide.md

```diff
@@ -132,6 +132,53 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 ## **Implementation**
 
 This section describes some noteworthy details on how certain features are implemented.
+### Resident-Room allocation feature 
+
+The allocation feature is facilitated by `ResidentRoom` which is a pair value of 
+valid`[Resident, Room]` that represents an existing resident allocated to a room. 
+It should be able to support the following operations:
+* `ResidentRoomList#getRoom()` - Returns the room assignment for a given resident.
+* `ResidentRoomList#getResident()` - Returns the resident allocated for a given room.
+
+A `ResidentRoomList` is a supplementary class that tracks all the `ResidentRoom` assignments. It should support
+the following operations. 
+* `ResidentRoomList#allocate()` - Adds a `ResidentRoom` new allocation 
+* `ResidentRoomList#deallocate()` - Removes an existing `ResidentRoom` allocation.
+* `ResidentRoomList#isAllocated()` - Checks if an allocation exists given the resident and room.
+* `ResidentRoomList#isRoomOccupied()` - Checks if a room is occupied. 
+* `ResidentRoomList#isResidentOccupied()` - Checks if a resident is allocated a room.
+
+There are two Resident-Room user allocation and deallocation commands, `alloc` and `dealloc` respectively.  
+
+The `alloc` command will do the following:
+
+Example: `alloc n/John Tan r/03-100`
+* Checks that `John Tan` exists.
+* Checks that `03-100` exists.
+* Checks that `John Tan` has not already been allocated to `03-100`.  
+* Checks that no other room is allocated to `John Tan`.  
+* Checks that there is room `03-100` is occupied by anyone. 
+
+    **If all the above is true,**
+* Set the `ROOM` of `John Tan` to be `03-100`.
+* Set the `OCCUPATION_STATUS` of a room to `Y`.
+
+  ![Activity Diagram of Allocation](images/ResidentRoomAllocationDiagram.png)
+
+The `dealloc` command will do the following:
```

Comment:
Perhaps it would be helpful here to link to the cascading impact on commands like `rdel`, `redit`, `odel` and `oedit`, which will need to be modified to block editing until deallocation happens. It would help explain the design decisions made on those commands.

Reply:
Added this point. Thank you.